modules/cudf/src/data_frame.ts

```diff
@@ -1588,34 +1588,31 @@ export class DataFrame<T extends TypeMap = any> {
    */
   sum<P extends keyof T>(subset?: (keyof T)[], skipna = true, memoryResource?: MemoryResource) {
     subset = (subset == undefined) ? this.names as (keyof T)[] : subset;
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    const containsIntType   = subset.some((name) => {
-      return (
-        [new Int8, new Int16, new Int32, new Int64].some(t => this.get(name).type.compareTo(t)));
-    });
-    const containsUintType  = subset.some((name) => {
-      return ([new Uint8, new Uint16, new Uint32, new Uint64].some(
-        t => this.get(name).type.compareTo(t)));
-    });
-    const containsFloatType = subset.some((name) => {
-      return ([new Float32, new Float64].some(t => this.get(name).type.compareTo(t)));
+
+    if (subset.length == 0) { throw new Error('Unable to calculate sum of empty DataFrame'); }
+
+    const floatTypes = [new Float32, new Float64];
+    const intTypes   = [new Int8, new Int16, new Int32, new Int64];
+    const UintTypes  = [new Uint8, new Uint16, new Uint32, new Uint64];
+
+    const firstSeriesDtype           = this.get(subset[0]).type;
+    const isFirstSeriesFloatingPoint = floatTypes.some(t => t.compareTo(firstSeriesDtype));
+
+    const sums = subset.map((name) => {
+      if (isFirstSeriesFloatingPoint) {
+        if (!floatTypes.some(t => this.get(name).type.compareTo(t))) {
```

Comment:
With the new Arrow v4.0.0 this API changed from `type.compareTo()` to a utility function `compareTypes()`: 
https://github.com/rapidsai/node-rapids/commit/1582c01381045a13f9a16a8d4c7f4db00e24d46c#diff-749cd88028943c88bd04f443e9453b52921f518db1cc04cfa0e84fa412a7a123R815

Reply:
Spoke with Paul about this, not needed since we are just going to infer using the type system.